test/TestServerProjectsLowLevel/dpg_update1/Generated/ParamsClient.cs

```diff
@@ -187,6 +187,68 @@ public virtual Response PutRequiredOptional(string requiredParam, string optiona
             }
         }
 
+        /// <summary> POST a JSON or a JPEG. </summary>
+        /// <param name="content"> The content to send as the body of the request. </param>
+        /// <param name="contentType"> Body Parameter content-type. Allowed values: &quot;application/json&quot; | &quot;image/jpeg&quot;. </param>
+        /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
+        /// <remarks>
+        /// Schema for <c>Request Body</c>:
+        /// <code>{
+        ///   url: string (required)
+        /// }
+        /// </code>
+        /// 
+        /// </remarks>
+        public virtual async Task<Response> PostParametersAsync(RequestContent content, ContentType contentType, RequestContext context = null)
+        {
+            Argument.AssertNotNull(content, nameof(content));
```

Comment:
Why are we validating a body parameter here?  We should only be validating path parameters.

Reply:
No, in the latest discussion we have agreed that if entire body object is exposed in a public API and that object is marked as required like this we should assert not null on that object, but not its properties. 

@m-nash , correct me if I'm wrong

Reply:
@AlexanderSher, we updated the guidelines to reflect the new guidance: https://azure.github.io/azure-sdk/general_implementation.html#general-params-server-validation

Can you please share where the "latest discussion" information happened?

Reply:
@AlexanderSher do you have any more information about this?

Reply:
I think we should not validate content as https://azure.github.io/azure-sdk/dotnet_introduction.html#parameter-validation. Created issue here: https://github.com/Azure/autorest.csharp/issues/2155